packages/discord.js/src/managers/GuildBanManager.js

```diff
@@ -199,6 +199,40 @@ class GuildBanManager extends CachedManager {
     await this.client.rest.delete(Routes.guildBan(this.guild.id, id), { reason });
     return this.client.users.resolve(user);
   }
+
+  /**
+   * Ban up to 200 users from a guild, and optionally delete previous messages sent by the banned users.
+   * @param {Collection<Snowflake, User>|UserResolvable[]} users The users to ban
+   * @param {BanBulkOptions} options The options for bulk banning users
+   * @returns {Promise<BanBulkResult>} Returns an object with `bannedUsers` key containing the IDs of the banned users
+   * and the key `failedUsers` with the IDs that could not be banned or were already banned.
+   * @example
+   * // Bulk ban users by ids (or with user/guild member objects)
+   * guild.bans.bulkCreate(['84484653687267328'], { deleteMessageSeconds: 7 * 24 * 60 * 60 })
+   *   .then(result => {
+   *     console.log(`Banned ${result.bannedUsers.length} users, failed to ban ${result.failedUsers.length} users.`)
+   *   })
+   *   .catch(console.error);
+   */
+  async bulkCreate(users, options) {
+    if (!users) throw new Error('Users array or collection must be provided');
+    if (typeof options !== 'object') throw new DiscordjsTypeError(ErrorCodes.InvalidType, 'options', 'object', true);
+
+    if (!Array.isArray(users) && !(users instanceof Collection)) {
+      throw new Error('Option "users" is not an array or collection');
+    }
+
+    let userIds = users instanceof Collection ? [...users.keys()] : users.map(user => user.id ?? user);
+    if (userIds.length === 0) throw new Error('Option "users" array or collection is empty');
+    if (userIds.length > 200) {
+      throw new Error('A maximum of 200 users can be passed as "users" parameter');
+    }
```

Comment:
These errors should probably be defined like the rest of the errors in our custom error classes